Sign in screen

joelgroomer opened the pull request:
# Description

Makes the Sign-In screen functional.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Manual tests
- [x] Automated tests

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

joelgroomer committed:
Added SignInScreen storyboard

joelgroomer committed:
Added SignInScreen storyboard

joelgroomer committed:
Add tap recognizers to sign in screen labels

joelgroomer committed:
Add tap recognizers to sign in screen labels

joelgroomer committed:
styled sign in button and added (blank) action for it

joelgroomer committed:
styled sign in button and added (blank) action for it

joelgroomer committed:
Secured password text

joelgroomer committed:
Secured password text

joelgroomer committed:
Wired up the SignIn screen to the UserController
Enabled ability to sign in. Remembers previous sign-ins via the token.

joelgroomer committed:
Merge remote-tracking branch 'origin/sign-in-screen' into sign-in-screen

joelgroomer committed:
Added missing Content-Type flags to HTTP requests

Keffury1 commented on ExpressWash/ExpressWash/Assets.xcassets/hand-washing-car.imageset/Contents.json:
```diff
@@ -0,0 +1,21 @@
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
Cool looking image.

Keffury1 commented on ExpressWash/ExpressWash/ColorPalette.md:
```diff
@@ -0,0 +1,7 @@
+Dark blue: #577399
+Light blue: #BDD5EA
+Light gray: #E5E5F3
+Red-Orange: #FE5F55 
```
Great color scheme  to work from

Keffury1 commented on ExpressWash/ExpressWash/Info.plist:
```diff
@@ -41,15 +41,15 @@
 	<string>LaunchScreen</string>
 	<key>UIMainStoryboardFile</key>
-	<string>Main</string>
+	<string>SignInScreen</string>
```
Smart to change the launch screen

Keffury1 commented on ExpressWash/ExpressWash/WasherController.swift:
```diff
@@ -171,6 +172,7 @@ extension WasherController {
             .appendingPathComponent(washer.stringID)
         var request = URLRequest(url: requestURL)
         request.httpMethod = "PUT"
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
```
Good thinking setting this value as above

Keffury1 commented on ExpressWash/SignInViewController.swift:
```diff
@@ -0,0 +1,83 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        btnSignIn.layer.cornerRadius = 5.0
```
Perhaps this could go in a function for future editing?

Keffury1 commented on ExpressWash/SignInViewController.swift:
```diff
@@ -0,0 +1,83 @@
+
+        // check to see if a valid token is already available
+        if UserController.shared.token != nil {
+            // TODO: after web team gives info about checking token validity, add it here
```
TODO shouldn't be in here

Keffury1 commented on ExpressWash/SignInViewController.swift:
```diff
@@ -0,0 +1,83 @@
+            }
+        }))
+        alert.addAction(UIAlertAction(title: "Cancel", style: .destructive, handler: { _ in }))
+        present(alert, animated: true, completion: nil)
```
Cool functionality. Function wording was a bit tricky.

Keffury1 commented on ExpressWash/SignInViewController.swift:
```diff
@@ -0,0 +1,83 @@
+                                          preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+            present(alert, animated: true, completion: nil)
+            return
```
I did the same in one of my PRs. We may want to customize alerts if we have time later

joelgroomer committed:
Bearer Token
Added functionality to validate and utilize the bearer token. Also added a way to re-authenticate if the token expires.

Keffury1 commented on ExpressWash/ExpressWash/UserController+authenticate.swift:
```diff
@@ -82,4 +85,83 @@ extension UserController {
+                }
+            }
+            completion(false)
+        }
```
This will be very useful!

Keffury1 commented on ExpressWash/ExpressWash/UserController+authenticate.swift:
```diff
@@ -82,4 +85,83 @@ extension UserController {
+        }
+
+        // if the user did not save their password...
+
```
Again solid commenting.

Keffury1 commented on ExpressWash/ExpressWash/UserController+authenticate.swift:
```diff
@@ -82,4 +85,83 @@ extension UserController {
+            sender.dismiss(animated: true, completion: nil)
+        }))
+
+        sender.present(alert, animated: true, completion: nil)
```
Looks like the meat of the changes here. All looks good.
